java-checks/src/main/java/org/sonar/java/checks/SwitchAtLeastThreeCasesCheck.java

```diff
@@ -20,34 +20,40 @@
 package org.sonar.java.checks;
 
 import org.sonar.check.Rule;
-import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.JavaFileScannerContext;
-import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
 import org.sonar.plugins.java.api.tree.CaseGroupTree;
 import org.sonar.plugins.java.api.tree.SwitchStatementTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import java.util.List;
 
 @Rule(key = "S1301")
-public class SwitchAtLeastThreeCasesCheck extends BaseTreeVisitor implements JavaFileScanner {
+public class SwitchAtLeastThreeCasesCheck extends IssuableSubscriptionVisitor {
```

Comment:
I think this move is a good one, as it will make the rule more performant. However, I would perfer having it in a 2nd distinct commit.
1. fix the FP
2. migrate from BaseTreeVisitor to IssuableSubscriptionVisitor

Ultimately, I would then keep the two commits as distinct when merging!